pkg/controllers/scheduling/schedule.go

```diff
@@ -253,19 +269,51 @@ func getWeights(defaultWeight map[string]int32, placement *clusterapiv1alpha1.Pl
 	}
 }
 
-func mergeWeights(defaultWeight map[string]int32, customizedWeight []clusterapiv1alpha1.PrioritizerConfig) map[string]int32 {
-	weights := make(map[string]int32)
+func mergeWeights(defaultWeight map[clusterapiv1alpha1.ScoreCoordinate]int32, customizedWeight []clusterapiv1alpha1.PrioritizerConfig) map[clusterapiv1alpha1.ScoreCoordinate]int32 {
+	weights := make(map[clusterapiv1alpha1.ScoreCoordinate]int32)
 	// copy the default weight
-	for k, v := range defaultWeight {
-		weights[k] = v
+	for sc, w := range defaultWeight {
+		weights[sc] = w
 	}
-	// override the default weight
+
 	for _, c := range customizedWeight {
-		weights[c.Name] = c.Weight
+		if c.ScoreCoordinate != nil {
+			weights[*c.ScoreCoordinate] = c.Weight
+		} else {
+			// override default weight
```

Comment:
please expand this comment indicating that this is for compatibility with pre-existing data before the scorecoordinate field was added.

Reply:
updated.